src/index.ts

```diff
@@ -361,6 +361,24 @@ export class ProjectConfig {
 
     return true;
   }
+
+  /**
+   * Generate a JSON string serialization of this configuration. File paths
+   * will be relative to root.
+   */
+  toJSON(): string {
+    const relative = (p: string) => p ? path.relative(this.root, p) : undefined;
```

Comment:
I'm checking for undefined mostly. But nulls and empty strings too, hence just the falsey check. I wanted to return undefined for those so that they don't serialize at all in the JSON.